src/consensus/pbft/libbyz/Pre_prepare.cpp

```diff
@@ -291,12 +291,21 @@ bool Pre_prepare::pre_verify()
 #ifdef SIGN_BATCH
     if (is_signed())
     {
-      if (!pbft::GlobalState::get_node()
-             .get_principal(sender)
-             ->verify_signature(
-               rep().digest.digest(),
-               rep().digest.digest_size(),
-               (const char*)get_digest_sig().data()))
+      auto sender_principal =
+        pbft::GlobalState::get_node().get_principal(sender);
+
+      if (
+        !sender_principal->has_certificate_set() &&
+        pbft::GlobalState::get_node().f() == 0)
+      {
+        // we have not configured this node yet
```

Comment:
I think the comment should be more explicit here? e.g. "Do not verify signature of first pre-prepares since node certificate required for verification is contained in the pre-prepare requests".